cg/meta/workflow/taxprofiler.py

```diff
@@ -18,3 +34,85 @@ def __init__(
         pipeline: Pipeline = Pipeline.TAXPROFILER,
     ):
         super().__init__(config=config, pipeline=pipeline)
+        self.root_dir = config.taxprofiler.root
+
+    @property
+    def root(self) -> str:
+        return self.root_dir
+
+    @property
+    def fastq_handler(self):
+        return TaxprofilerFastqHandler
+
+    def get_case_config_path(self, case_id):
+        return NextflowAnalysisAPI.get_case_config_path(case_id=case_id, root_dir=self.root_dir)
+
+    @staticmethod
+    def build_samplesheet_content(
+        case_id: str,
+        fastq_r1: List[str],
+        fastq_r2: List[str],
+        instrument_platform: str,
+        fasta: Optional[str],
+    ) -> Dict[str, List[str]]:
+        """Build samplesheet headers and lists"""
+        try:
+            TaxprofilerSample(
+                sample=case_id,
+                fastq_r1=fastq_r1,
+                fastq_r2=fastq_r2,
+                instrument_platform=instrument_platform,
+                fasta=fasta,
+            )
+        except ValidationError as error:
+            LOG.error(error)
+            raise ValueError
+
+        samples_full_list: list = []
+        instrument_full_list: list = []
+        fasta_full_list: list = []
```

Comment:
`List[str]` maybe?

Reply:
Do you mean for all of them?

Reply:
Yes, they are all lists with `string` types.

I get this error after applying the review suggestion when running `cg worfklow taxprofiler config-case richurchin`

```
Case richurchin exists in status db
Generating samplesheet
Fetching files from bundle ACC10629A1
Fetching files with tags in [fastq]
{'sample': ['richurchin', 'richurchin', 'richurchin', 'richurchin'], 'run_accession': ['richurchin', 'richurchin', 'richurchin', 'richurchin'], 'instrument_platform': [<SequencingPlatform.illumina: 'ILLUMINA'>, <SequencingPlatform.illumina: 'ILLUMINA'>, <SequencingPlatform.illumina: 'ILLUMINA'>, <SequencingPlatform.illumina: 'ILLUMINA'>], 'fastq_1': ['/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L001_R1_001.fastq.gz', '/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L002_R1_001.fastq.gz', '/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L003_R1_001.fastq.gz', '/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L004_R1_001.fastq.gz'], 'fastq_2': ['/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L001_R2_001.fastq.gz', '/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L002_R2_001.fastq.gz', '/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L003_R2_001.fastq.gz', '/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L004_R2_001.fastq.gz'], 'fasta': ['', '', '', '']}
Traceback (most recent call last):
  File "/home/proj/stage/bin/miniconda3/envs/S_main/bin/cg", line 8, in <module>
    sys.exit(base())
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/cli/workflow/taxprofiler/base.py", line 41, in config_case
    analysis_api.config_case(case_id=case_id, instrument_platform=instrument_platform, fasta="")
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/meta/workflow/taxprofiler.py", line 116, in config_case
    case_id=case_id, instrument_platform=instrument_platform, fasta=fasta
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/meta/workflow/taxprofiler.py", line 105, in write_samplesheet
    case_id=case_id, root_dir=self.root_dir
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/meta/workflow/nextflow_common.py", line 132, in create_samplesheet_csv
    outfile.write(",".join(headers))
TypeError: sequence item 3: expected str instance, list found
```

I think if I change the function `create_samplesheet_csv` in `nextflow_common.py`, this will create issues in Rnafusion.

Reply:
This method should return the following dictionary:

`def build_sample_sheet_content() -> Dict[str, List[str]]:`

you're returning a Dict[str, str]:
```
        samplesheet_content: Dict[str, str] = {
            NFX_SAMPLE_HEADER: samples_full_list,
            TAXPROFILER_RUN_ACCESSION: samples_full_list,
            TAXPROFILER_INSTRUMENT_PLATFORM: instrument_full_list,
            NFX_READ1_HEADER: fastq_r1,
            NFX_READ2_HEADER: fastq_r2,
            TAXPROFILER_FASTA_HEADER: fasta_full_list,
        }
        return samplesheet_content
```

Reply:
Thank you for all the suggestions and comments ⭐ 

I updated the function to return a `Dict[str, List[str]]` but I was getting the same error.
 
I think that the error was raised from the definition of `TAXPROFILER_SAMPLE_SHEET_HEADERS` in `cg/constants/taxprofiler.py`. It was a `list` but it needs to be a `list[str]`.

I changed back to the definition from the initial commit. Not sure what would be an ideal definition here.